### DISCUSSION TITLE
How can I Instantiate a Gameobject at runtime like zenject's sub container

### DISCUSSION THREAD

**Discussion by cowbear6598** [hadashiA/VContainer#473] [Q&A] [answered]
like title, I have CharacterView (MonoBehaviour)、CharacterMoveHandler (plain C#) and the handler need the view.
In zenject I can use Installer to do this, but in vcontainer how can I do?

I only thought a way that I create CharacterPresenter (plain C#) and CharacterRepository (plain C#), when instantiate a character I will put view and handler in repository, and then presenter get all character from repository and run all handlers' tick

Is there have more easy way to do this?

**Comment by SimonNordon4** [comment, marked as answer]
``` csharp
[SerializedField]
private CharacterView characterPrefab;

builder.RegisterFactory<CharacterMoveHandler >(container =>
{
    return () =>
    {
        var CharacterView = container.Instantiate(characterPrefab);
        var moveHandler = new MoveHandler(CharacterView);
    };
}, Lifetime.Scoped);

```
Something like this should work.

**Reply by cowbear6598** [reply]
I notice 1.13.0 fixed this bug, https://github.com/hadashiA/VContainer/pull/452, I will try again later
